src/spi_comms.c

```diff
@@ -0,0 +1,357 @@
+#include "spi_comms.h"
+
+// SPI data from PAY-SSM
+volatile uint8_t spi_first_byte = 0;
+volatile uint8_t spi_second_byte = 0;
+
+volatile uint8_t spi_spdr_tx_buffer = 0;    // holds data to be shifted into SPDR, for SPI transfer
+volatile uint8_t spi_frame_number = 0;      // byte number of data trasmission
+
+volatile uint8_t spi_status_byte = 0;

```

Comment:
Don't need any global variables

Reply:
Since these variables are modified by 1 function and used by another function, should I be passing them around by function return?

Reply:
You should only need them in one function which is `opt_loop()`